### PR TITLE
Add the application CAM360

### PR DESCRIPTION
**Pull request by AleziaKurdis**
This PR adds the application CAM360
(Formerly known as RPO360, originallly created by Hight Fidelity, but with significant modifications)

The Original RPO360 application _(Yeah, "**R**eady **P**layer **O**ne" 360 camera)_ was a camera that you could throw in the air to capture a Spherical snapshot.
Once captured it was possible to rez a shere with the snapshot on it to let people see it.
(to be able to do this, it was sharing the image on the high fildelity metaverse server.)

The original application was broken because this image sharing doesn't exist anymore, (the service on the current "Directory" server has never been re-coded for some actual reasons). It was also broken because of this bug about the method "Entities.getActionIDs": https://github.com/overte-org/overte/issues/176

This new version, re-coded at 75%, works a bit differently.
There is now 2 modes:
- The "**Throw mode**": Rez the cam, grab it, then toss it high in the sky to capture a 360 snapshot.
image

- The "**Position mode**": Position then click to capture a360 snapshot. Th capture can be triggered on the tablet but also by pressing the thumbstick oh the right controller.
image

Since we can't share anymore the picture, the preview is now only available to the user. It came as a 360° media viewer on the tablet, letting you see the last snapshot, based on the tablete rotation (in HMD) or the user camera rotation (in Desktop).
The ui has been rebuilt in HTML to have this possibility. (All the code line including the "marzipano" library used for that, is licensed Apache 2.0)

The throwable camera had to be rezzed as a local entity (instead of an avatar entity) so only the user ca see it. (It was the only way I found to workaround the broken "Entities.getActionIDs" issue.)
The positionable camera is still an avatar entity, which allow to use the camera with participation of other people.

Even if we already have a similar capture capability on the "specator camera", this one has the the feature to see immediately the result of the capture. And the throwable is always fun to use during events.

The camera model has been remade completely, (decreasing the number of polygons of 50%, and removing at least one material.)
image

### CODE REVIEW

File: applications/cam360/cam360.js [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+            });
+        }
+    }
+
+    var controllerType = "Other";
+    function registerButtonMappings() {
+        var VRDevices = Controller.getDeviceNames().toString();
+        if (VRDevices) {
+            if (VRDevices.indexOf("Vive") !== -1) {
+                controllerType = "Vive";
```
Comment (ksuprynowicz) [inline applications/cam360/cam360.js:588]: We will need to update this so it works with other devices too, but we can do it in the separate commit.

**Review by ksuprynowicz — CHANGES_REQUESTED** [review]
Everything looks fine in my opinion, excep channel name. Shouldn't it end with `overte.org` instead?

File: applications/cam360/cam360.js (hunk 1 above)
Comment (JulianGro) [reply applications/cam360/cam360.js:588]: Just for reference, "Vive" probably means any OpenVR device in practice, because we still use legacy bindings (from the time when the Vive was the only OpenVR device) that SteamVR provides backwards compatibility for.

File: applications/cam360/cam360.html [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+    <head>
+        <meta charset="UTF-8">
+        <script src="resources/marzipano/marzipano.js" ></script>
+        <script>
+
+            //Paths
+            var thisPageName = "cam360.html";        
+            var currentPath = window.location.protocol + "//" + window.location.host + window.location.pathname;
+            var ROOTPATH = currentPath.replace(thisPageName, ""); 
+            var channel = "cam360.ak.overte.com";
```
[the remark this answers is not in the document]
Comment (AleziaKurdis) [reply applications/cam360/cam360.html]: I changed it for : "org.overte.applications.cam360"

File: applications/cam360/cam360.js (hunk 1 above)
Comment (AleziaKurdis) [reply applications/cam360/cam360.js:588]: I was doing the same lecture of this. If this becomes a problem for any new devices we will have to re-adjust this (with probably a lot of other scripts too)

**Review by ksuprynowicz — APPROVED** [review]
